elements/lisk-validator/src/keywords/data_type.ts

```diff
@@ -75,13 +80,40 @@ const compile = (
 		_parentDataProperty?: string | number,
 		_rootData?: object,
 	): boolean => {
-		if (value === 'boolean') return isBoolean(data);
-		if (value === 'bytes') return isBytes(data as Buffer);
-		if (value === 'string') return isString(data);
-		if (value === 'uint32') return isUInt32(data);
-		if (value === 'uint64') return isUInt64(data);
-		if (value === 'sint32') return isSInt32(data);
-		if (value === 'sint64') return isSInt64(data);
+		if (value === 'boolean') {
+			return isBoolean(data);
+		}
+		if (value === 'bytes') {
+			if (!isBytes(data as Buffer)) {
+				return false;
+			}
+			const parent = parentSchema as KVPair;
+			if (typeof parent.minLength === 'number') {
```

Comment:
When you can validate a keyword value with another keyword then its recommended to use `macro` instead of `compile`. But here is the mixup of both, never seen such a case, so not sure which is Ajv recomendation to handle this case. So we can keep it same for now.